velox/tool/trace/PartitionedOutputReplayer.cpp

```diff
@@ -0,0 +1,167 @@
+          int64_t /*inSequence*/,
+          std::vector<int64_t> /*remainingBytes*/) {
+        result->setValue(std::move(pages));
+      }));
+  auto future = std::move(semiFuture).via(executor);
+  future.wait(std::chrono::seconds{10});
+  VELOX_CHECK(future.isReady());
+  return std::move(future).value();
+}
+} // namespace
+
+std::vector<std::vector<std::unique_ptr<folly::IOBuf>>> getAllData(
+    const std::shared_ptr<exec::OutputBufferManager>& bufferManager,
+    const std::string& taskId,
+    uint32_t numPartitions,
+    folly::Executor* executor) {
+  std::vector<std::thread> consumerThreads;
+  std::vector<std::vector<std::unique_ptr<folly::IOBuf>>> partitionedResults;
+  consumerThreads.reserve(numPartitions);
+  partitionedResults.reserve(numPartitions);
+  partitionedResults.resize(numPartitions);
+  for (uint32_t i = 0; i < numPartitions; i++) {
+    consumerThreads.push_back(std::thread([&, partition = i]() {
+      bool finished{false};
+      while (!finished) {
+        std::vector<std::unique_ptr<folly::IOBuf>> pages;
+        {
+          pages = getData(
+              bufferManager,
+              taskId,
+              partition,
+              partitionedResults[partition].size(),
+              executor);
+        }
+        for (auto& page : pages) {
+          if (page) {
+            partitionedResults[partition].push_back(std::move(page));
+          } else {
+            // Null page indicates this buffer is finished.
+            bufferManager->deleteResults(taskId, partition);
+            finished = true;
+          }
+        }
+      }
+    }));
+  }
+
+  for (auto& thread : consumerThreads) {
+    thread.join();
+  }
+  return partitionedResults;
+}
+
+PartitionedOutputReplayer::PartitionedOutputReplayer(
+    const std::string& rootDir,
+    const std::string& taskId,
+    const std::string& nodeId,
+    const int32_t pipelineId,
+    const std::string& operatorType)
+    : OperatorReplayerBase(rootDir, taskId, nodeId, pipelineId, operatorType),
+      originalNode_(dynamic_cast<const core::PartitionedOutputNode*>(
+          core::PlanNode::findFirstNode(
+              planFragment_.get(),
+              [this](const core::PlanNode* node) {
+                return node->id() == nodeId_;
+              }))) {
+  VELOX_CHECK_NOT_NULL(originalNode_);
+}
+
+RowVectorPtr PartitionedOutputReplayer::run() {
+  auto task = Task::create(
+      "local://partitioned-output-replayer",
+      core::PlanFragment{createPlan()},
+      0,
+      createQueryContext(queryConfigs_, executor_.get()),
+      Task::ExecutionMode::kParallel);
+  task->start(maxDrivers_);
+
+  auto partitionedResults = getAllData(
+      bufferManager_,
+      task->taskId(),
+      originalNode_->numPartitions(),
+      executor_.get());
+  common::testutil::TestValue::adjust(
+      "facebook::velox::tool::PartitionedOutputReplayer::run",
+      &partitionedResults);
+  return nullptr;
```

Comment:
Instead of capture though test value, can we convert to std::vector<RowVectorPtr> as output? thanks!